### PR TITLE
Fixed Chat-render-hint & Infinite loading bug

### PR DESCRIPTION
**Pull request by derech1e**
fixes CCBlueX/LiquidBounce1.8-Issues#3725

fixes CCBlueX/LiquidBounce1.8-Issues#3815
fixes probably also CCBlueX/LiquidBounce1.8-Issues#3239
closes CCBlueX/LiquidBounce1.8-Issues#3500

### CODE REVIEW

File: 1.8.9-Forge/src/main/java/net/ccbluex/liquidbounce/features/module/modules/misc/NameProtect.java [hunk 1]
```diff
@@ -31,16 +34,15 @@ public void onText(final TextEvent event) {
             return;
 
         for (final FriendsConfig.Friend friend : LiquidBounce.fileManager.friendsConfig.getFriends())
-            event.setText(StringUtils.replace(event.getText(), friend.getPlayerName(), ColorUtils.translateAlternateColorCodes(friend.getAlias()) + "§f"));
```
Comment (superblaubeere27) [inline 1.8.9-Forge/src/main/java/net/ccbluex/liquidbounce/features/module/modules/misc/NameProtect.java:34]: StringUtils is faster than `String.replace`

File: 1.8.9-Forge/src/main/java/net/ccbluex/liquidbounce/utils/misc/StringUtils.java [hunk 2]
```diff
@@ -18,31 +18,4 @@ public static String toCompleteString(final String[] args, final int start) {
...
-                    return string;
-
-                return stringBuilder.toString();
-            }
-
-            stringBuilder.replace(start, start + searchCharsLength, replaceChars);
-        }
-
-        return stringBuilder.toString();
-    }
```
Comment (superblaubeere27) [inline 1.8.9-Forge/src/main/java/net/ccbluex/liquidbounce/utils/misc/StringUtils.java:47]: `>:C`

File: 1.8.9-Forge/src/main/java/net/ccbluex/liquidbounce/features/module/modules/misc/NameProtect.java (hunk 1 above)
Comment (derech1e) [reply 1.8.9-Forge/src/main/java/net/ccbluex/liquidbounce/features/module/modules/misc/NameProtect.java:34]: Okay, for me it lags as fuck: https://streamable.com/rl8sj3
My friendlist: https://pastebin.com/gUaFhBzm
It is realyl annoying when you tab your friendlist and load it forever.
Do you have a better solution?

Comment (derech1e) [reply 1.8.9-Forge/src/main/java/net/ccbluex/liquidbounce/features/module/modules/misc/NameProtect.java:34]: Try it on your own.

**Pull request closed without merging**